src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +353,288 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
+
+        ______TS("success: giver is instructor; show comment to student; comment is visible to recipient");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is instructor; show comment to student; comment is visible to teammates");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student3InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is instructor; comment is visible to giver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedComment = FeedbackResponseCommentAttributes.builder()
+                .withCommentGiver("student4InCourse1@gmail.tmt")
+                .withCommentGiverType(FeedbackParticipantType.STUDENTS)
+                .withVisibilityFollowingFeedbackQuestion(false)
+                .withCourseId("idOfTypicalCourse1")
+                .withFeedbackSessionName("First feedback session")
+                .withShowCommentTo(Arrays.asList(FeedbackParticipantType.STUDENTS))
+                .build();
+
+        ______TS("failure: giver is student; show comment to student; comment is not"
+                + "visible to instructor");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; show comment to student; comment is visible to giver");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; show comment to student; comment is visible to recipient");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is student; show comment to student; comment is visible to teammates");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student3InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedComment = FeedbackResponseCommentAttributes.builder()
```

Comment:
same here - better to just create another comment instead of mutating existing one